contribs/drt/src/main/java/org/matsim/contrib/drt/analysis/DrtEventSequenceCollector.java

```diff
@@ -217,11 +247,85 @@ public void handleEvent(PersonMoneyEvent event) {
 							+ " or similar. Terminating.", event.getPurpose(), event.getTransactionPartner());
 
 			drtFarePersonMoneyEvents.add(event);
-			PerformedRequestEventSequence sequence = performedRequestSequences.get(
+			EventSequence sequence = sequences.get(
 					Id.create(event.getReference(), Request.class));
 			if (sequence != null) {
 				sequence.drtFares.add(event);
 			}
 		}
 	}
+
+	/*
+	 * This function is helper that finds a PersonDepartureEvent for a given
+	 * DrtRequestSubmittedEvent. This means that the departure has happened before
+	 * submission, which is the usually case for instant requests.
+	 */
+	private Optional<PersonDepartureEvent> popDepartureForSubmission(DrtRequestSubmittedEvent event) {
```

Comment:
Suggestion: return simply a (nullable) PersonDepartureEvent? `Optional` is intended for public API to make it more expressive, whereas here it's a private method and the results is consumed inside this class.